state/postgresql/postgresdbaccess.go

```diff
@@ -274,7 +270,7 @@ func (p *postgresDBAccess) ensureStateTable(stateTableName string) error {
 		p.logger.Info("Creating PostgreSQL state table")
 		createTable := fmt.Sprintf(`CREATE TABLE %s (
 									key text NOT NULL PRIMARY KEY,
-									value json NOT NULL,
+									value text NOT NULL,
```

Comment:
Thanks for fixing postgresql to pass the conformance tests ... can you explain why the type for the column needs to be changed to `text` though? It looks like elsewhere in the `setValue` the code explicitly marshals to `json`. What was the problem that required this type change?

Reply:
It looks like both postgresql and mysql cannot accept the following payload as a json type.


https://github.com/dapr/components-contrib/blob/master/tests/conformance/state/state.go#L70

```go
		{
			key:                  fmt.Sprintf("%s-trx-bytes", key),
			value:                []byte{0x1},
			expectedReadResponse: []byte{0x1},
			transactionOnly:      true,
			transactionGroup:     1,
		},
```

otherwise, the following error occurs:

```
Expected nil, but got: &pgconn.PgError{Severity:"ERROR", Code:"22P02", Message:"invalid input syntax for type json", Detail:"Token "testValue2" is invalid.", Hint:"", Position:0, InternalPosition:0, InternalQuery:"", Where:"JSON data, line 1: testValue2", SchemaName:"", TableName:"", ColumnName:"", DataTypeName:"", ConstraintName:"", File:"jsonfuncs.c", Line:621, Routine:"json_ereport_error"
```

Reply:
I see, thanks for explaining the problem. Given the expectation that the component expects the raw byte blob to be returned as-is on the Get call, I think change of the field to `text` is reasonable, and there is no expectation in the postgres state component itself that the value column will be manipulated via JSON inspection, so it doesn't seem to matter. 

I have other concerns about the attempt to marshal as JSON if the intent of the conformance tests is that the state stores are binary-preserving:
https://github.com/dapr/components-contrib/blob/418fc726f5501239bcf034ae09a4a29e040ba758/state/postgresql/postgresdbaccess.go#L101
but that should be tracked as part of the broader discussion in https://github.com/dapr/dapr/issues/3533#issuecomment-911789176 and not here.

Reply:
Instead of converting to JSON, is there a way to handle the special case for ""? There are advantages to keep the structure as JSON since it can allow queries in the near future. BTW, JSONB seems to be the right choice here: http://www.silota.com/docs/recipes/sql-postgres-json-data-types.html

Regarding "", it might be better off to store it as null - so making this column nullable could allow that.

Reply:
Also, byte[] should also be handled in a special way. In cosmosDB, there is a new property (would be a new column here) to specify that the value is binary and should be base64 decoded on read.